### ISSUE TITLE
Parsing mod

### ISSUE THREAD

**Issue by tramyn** [sbmlteam/jsbml#154]
When I call parseFormula to create an ASTNode using 'V0 % 2', it changes the math to piecewise(V0-2*ceil(V0/2),  xor (V0 < 0, 2 < 0), V0-2*floor(V0/2)). I have an SBMLDocument L3V2, is it possible to preserve the original math?

**Comment by niko-rodrigue** [comment]
Yes, indeed we should use the 'rem' operator for L3V2 document, I will have a look at it.

**Comment by niko-rodrigue** [comment]
The problem here is that the current formula parsers ignore everything of the SBML Level and Version. I have added a fix in c178bf57d1a46fdb61b5a60458baf56c30d17380 where you are able to create a parser and tell it which level and version to use when constructing the ASTNode and I added a fix for modulo if you set it to L3V2

Here is a code sample that use this feature:
 
```java
String modulo = "V0 % 2";
    
    try {
      FormulaParserLL3 parser = new FormulaParserLL3(new StringReader(""));
      parser.setLevel(3);
      parser.setVersion(2);
      
      ASTNode mod = ASTNode.parseFormula(modulo, parser);
      
       System.out.println(mod.toFormula());
      
      ASTNode rem = ASTNode.parseFormula("rem(V0, 2)");

      System.out.println(rem.toFormula());

    } catch (ParseException e) {
      e.printStackTrace();
    }
```

Would that work for you ? We could also add more `ASTNode.parseFormula` methods where you can pass the level and version as parameters so that you don't need to go through the steps of creating and initialising the parser yourself.

**Comment by tramyn** [comment]
Thanks @niko-rodrigue for this fix. This will work for me. Has this feature been deployed to Sonatype?
